### PR TITLE
Fix MarkdownEditor fullHeight support

### PR DESCRIPTION
**Pull request by stkao05**
Closes #

#### Changed

Currently, the MarkdownEditor's fullHeight prop doesn't take effect because the height change isn't applied to the outermost container element (`<fieldset>`). The PR fixes the issue by applying full height style to the the container element. 

Storybook has also been updated for demo purpose.

(Demo screen recording: after the fix, the MarkdownEditor correctly resizes to full height when fullHeight is turned on)

https://github.com/primer/react/assets/1556390/813c9ee5-45eb-4433-a902-91749eb2d60f

### Rollout strategy

- [x] Patch release
- [ ] Minor release
- [ ] Major release; if selected, include a written rollout or migration plan

### Merge checklist

- [ ] Added/updated tests
- [ ] Added/updated documentation
- [x] Added/updated previews (Storybook)
- [x] Changes are [SSR compatible](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#ssr-compatibility)
- [x] Tested in Chrome
- [x] Tested in Firefox
- [x] Tested in Safari
- [x] Tested in Edge

### CODE REVIEW

**Comment by cs25-esc** [comment]
Hi @stkao05 can you help to write a changeset for my PR https://github.com/primer/react/pull/3895 please?

File: .changeset/fluffy-months-invite.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
```
Comment (siddharthkp) [inline .changeset/fluffy-months-invite.md:1]: There are 2 changesets here, I imagine you only want the `patch` one?

**Review by siddharthkp — CHANGES_REQUESTED** [review]
Left a couple of comments

File: src/drafts/MarkdownEditor/MarkdownEditor.stories.tsx [hunk 2]
```diff
@@ -242,28 +244,30 @@ export const Playground = ({
...
-        mentionSuggestions={mentionables}
-        referenceSuggestions={references}
-        savedReplies={savedRepliesEnabled ? savedReplies : undefined}
-        required={required}
-        pasteUrlsAsPlainText={pasteUrlsAsPlainText}
-      >
-        <MarkdownEditor.Label visuallyHidden={hideLabel}>Markdown Editor Example</MarkdownEditor.Label>
-      </MarkdownEditor>
-      <p>Note: for demo purposes, files starting with &quot;A&quot; will be rejected.</p>
+      <div style={{height: '400px'}}>
```
[the remark this answers is not in the document]
Comment (stkao05) [reply src/drafts/MarkdownEditor/MarkdownEditor.stories.tsx:247]: I set the height of this element so we can test if the child `MarkdownEditor` component does extends to full height to its parent element height. 

But I agree it is a bit confusing from the code. I can add comment or remove the change if you think it isn't needed.

[the remark this answers is not in the document]
Comment (stkao05) [reply src/drafts/MarkdownEditor/MarkdownEditor.stories.tsx]: Remove the change for now.

**Comment by stkao05** [comment]
Thanks for reviewing. Just update the PR accordingly.

**Comment by siddharthkp** [comment]
@stkao05, There are some recent changes in `main` as well, can you merge latest `main` into your branch?

**Comment by stkao05** [comment]
> There are some recent changes to MarkdownEditor in `main` as well, can you merge latest `main` into your branch?

no problem. just did 👌

**Review by siddharthkp — APPROVED** [review]
This looks good! Will merge after CI passes :)

**Pull request merged**